frontend/src/components/Playback/OsdPlayer.vue

```diff
@@ -0,0 +1,354 @@
+                <v-icon>
+                  <i-mdi-fullscreen v-if="!$props.isFullscreen" />
+                  <i-mdi-fullscreen-exit v-else />
+                </v-icon>
+              </tooltip-button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </v-overlay>
+</template>
+
+<script lang="ts" setup>
+import { ref, onMounted, onBeforeUnmount, computed, watch } from 'vue';
+import IMdiClose from 'virtual:icons/mdi/close';
+import { useTimeoutFn } from '@vueuse/shared';
+import IMdiChevronDown from 'virtual:icons/mdi/chevron-down';
+import { BaseItemKind } from '@jellyfin/sdk/lib/generated-client';
+import { useVModel } from '@vueuse/core';
+import { playbackManagerStore, videoPlayerStore } from '@/store';
+import { mediaControls } from '@/store/videoPlayer';
+
+const emit = defineEmits<{
+  (e: 'toggleFullscreen'): void;
+  (e: 'update:modelValue', val: boolean): void;
+}>();
+
+const props = defineProps<{
+  modelValue: boolean;
+  isFullscreen: boolean;
+}>();
+
+const playbackManager = playbackManagerStore();
+const videoPlayer = videoPlayerStore();
+
+const overlay = useVModel(props, 'modelValue', emit);
+const subtitleSelectionButtonOpened = ref(false);
+const playbackSettingsButtonOpened = ref(false);
+const staticOverlay = computed(
+  () =>
+    playbackManager.isPaused ||
+    subtitleSelectionButtonOpened.value ||
+    playbackSettingsButtonOpened.value
+);
+
+watch(staticOverlay, (val) => {
+  if (val) {
+    timeout.stop();
+  } else {
+    timeout.start();
+  }
+});
+
+const timeout = useTimeoutFn(() => {
+  overlay.value = false;
+}, 5000);
+
+/**
+ * Shows the overlay on mouse move and starts the timeout to hide it, unless the overlay must stay static
+ */
+function handleMouseMove(): void {
+  if (!staticOverlay.value) {
+    overlay.value = true;
+    timeout.start();
+  }
+}
+
+const keyPressMappings: { [key: string]: () => void } = {
+  Spacebar: () => playbackManager.playPause(),
+  ' ': () => playbackManager.playPause(),
+  k: () => playbackManager.playPause(),
+  ArrowRight: () => playbackManager.skipForward(),
+  l: () => playbackManager.skipForward(),
+  ArrowLeft: () => playbackManager.skipBackward(),
+  j: () => playbackManager.skipBackward(),
+  f: () => {
+    if (playbackManager.currentlyPlayingMediaType === 'Video') {
+      emit('toggleFullscreen');
+    }
+  },
+  m: () => playbackManager.toggleMute()
+};
+
+/**
+ * Handles the key presses
+ */
+function handleKeyPress(e: KeyboardEvent): void {
+  if (keyPressMappings[e.key]) {
+    keyPressMappings[e.key]();
+  }
+}
+
+onMounted(() => {
+  window.addEventListener('mousemove', handleMouseMove);
+  window.addEventListener('touchend', handleMouseMove);
+  window.addEventListener('keyup', handleKeyPress);
+});
+
+onBeforeUnmount(() => {
+  window.removeEventListener('mousemove', handleMouseMove);
+  window.removeEventListener('touchend', handleMouseMove);
+  window.removeEventListener('keyup', handleKeyPress);
+});
```

Comment:
Vue 3 event binding changed, so you should be able to use the ``@`` syntax is components now: https://v3-migration.vuejs.org/breaking-changes/v-on-native-modifier-removed.html

Try that first please. In case you have no luck, replace this with https://vueuse.org/core/useEventListener/

Do the same with all the occurences in the code please (this is the only one I found)

Reply:
Ended up using the vueuse listeners cause the v-overlay isn't rendered or shown (if eager) and using event on its parent was too complicated.